Media Gallery component now receives allowable file types from File Uploader

gwharton opened the pull request:
### Description (*)
Extended allowable extensions for image type to include ico and apng (These are listed for the Favicon)

Magento Ui image uploader component now passes the list of allowable file types to the Media Gallery component. This will (optionally) override the default list of allowable file types obtained from di.xml. Previously the Media Gallery would just use the default set of allowable types as set in di.xml, even if the Image Uploader component was configured for a different set.

Implemented fallback thumbnail placeholder generation for Media Gallery when filetype is not supported by current Image Adapter. As no ico thumbnails can be generated by GD2, it shows the placeholder image instead when GD2 Image Adapter is selected.

Added a check to the uploader to skip resizing uploaded image for thumbnail generation if the image file extension is ico and the Image Adapter is GD2. 

Corrected error message style.

Added MFTF test.

### Related Pull Requests
Replaces the following PR
1. Closes #25007: ICO file not supported in media file uploader fixed, issue #23881

### Fixed Issues (if relevant)
1. Closes #23881: "Select from Gallery" option on media file uploader does not use correct allowable file types

### Manual testing scenarios (*)
As per original issue

### Questions or comments
Although ico is added to the list of allowable file types, and you are able to upload them, they are NOT supported by the GD2 Image Adapter which is the default. Therefore additional logic was required to skip thumbnail generation for the uploaded image, which although was handled gracefully on upload, caused an error message to be shown on the Media Gallery screen when ico files were uploaded. The error message is no longer shown.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

gwharton commented:
@magento give me test instance

magento-engcom-team commented:
Hi @gwharton. Thank you for your request. I'm working on Magento instance for you

magento-engcom-team commented:
Hi @gwharton, here is your [new Magento instance](https://pr-26980.instances.magento-community.engineering).
Admin access: [https://pr-26980.instances.magento-community.engineering/admin_7f58](https://pr-26980.instances.magento-community.engineering/admin_7f58)
Login: `eda376e7` Password: `648283ed150d`
Instance will be terminated in up to 3 hours.

gwharton commented:
For discussion, yes we are displaying a warning saying that the image is not supported, but actually, in terms of favicon, the image format is very much supported. It is only unsupported in terms of generating a thumbnail for the media browser.

The problem is, this is just the generic error message spewed out by the GD2 image adapter when it receives an unsupported image.

gwharton commented:
Functional Tests currently failing due to lack of ImageMagick php module in CI environment. Will investigate if this can be added, otherwise, I will remove the ImageMagic Image Adapter based tests.

Commit:
Media Gallery component, allowable file types.

Extended allowable extensions for image type to include ico and apng.

Magento Ui image uploader component now passes the list of allowable
file types to the Uploader component. This will override the default
list of allowable file types obtained from config if present.

Implemented fallback thumbnail placeholder generation for Media
Gallery when filetype is not supported by current Image Processor.

Added a check to the uploader to skip resizing uploaded image for
thumbnail generation if the image file extension is ico and the Image
Adapter is GD2.

Corrected error message style.

Added MFTF tests

lbajsarowicz commented on app/code/Magento/Cms/Block/Adminhtml/Wysiwyg/Images/Content/Uploader.php:
```diff
@@ -71,4 +71,18 @@ protected function _getMediaType()
+    {
+        if ($encodedExtensions = $this->getRequest()->getParam('allowed_extensions')) {
+            // phpcs:ignore Magento2.Functions.DiscouragedFunction
+            return(explode(" ", base64_decode($encodedExtensions)));
```
@piotrekkaminski This does not look safe for me. Could you take a look?

gwharton replied:
What would you recommend. I couldn't think of a better way of passing the list of file types into the media gallery component than this. Open to suggestions.
